changes/1502.feature.rst

```diff
@@ -0,0 +1,36 @@
+Improved the notification support in several ways:
+
+- Replaced the event-based handover of a single item from the notifiction
+  listener thread to the caller's thread with a Python Queue, for better
+  reliability.
+
+- Added methods 'connect()' and 'is_connected()' to the 'NotificationReceiver'
+  class. The init method of 'NotificationReceiver' no longer connects,
+  but the 'notifications()' method now calls 'connect()', so overall this is
+  compatible with the prior behavior.
+
+- Added exceptions 'NotificationConnectionError' and
+  'NotificationSubscriptionError'.
+
+- The 'NotificationReceiver.notifications()' method now continues running
+  when there are no notifications, and only ever returns when
+  'NotificationReceiver.close()' is called (by some other thread).
+
+- Added proper detection of STOMP connection loss if STOMP heartbeating is
+  enabled. The connection loss is surfaced by raising
+  'NotificationConnectionError' in 'NotificationReceiver.notifications()'.
+  This allows users to retry 'NotificationReceiver.notifications()' upon
+  connection loss.
+
+- Added a new public constant 'STOMP_MIN_CONNECTION_CHECK_TIME' that defines
+  the minimum time between checks for STOMP connection loss. The check time
+  increases from this minimum with as the heartbeat receive time gets larger.
+
+- Added the missing event methods to the internal '_NotificationListener'
+  class in case they are ever invoked (needed due to lazy importing of
+  stomp-py).
+
+- Added more log messages around STOMP connect / disconnect.
+
+- Documented the stomp-py exceptions that can be raised from public
+  zhmcclient methods.
```

Comment:
Add "Issue:" in a separate line so the issue link gets added there and not to the last item in the list.

Reply:
done